### PR TITLE
[FIX] survey : fix displaying progression when consulting results

### PR DESCRIPTION
**Pull request by abla001**
To reproduce
============

- Create a new Survey.
- Add questions under at least two sections.
- In Options, ensure that each section is displayed on its own page.
- Click "Share" and open the share link.
- Partially complete the survey and close the window before submitting.
- From the survey click "See Results". Then choose a response to a question from the partially completed answer.
- A traceback raised

Purpose
=======

On this condition :
https://github.com/odoo/odoo/blob/3602fdfc0ce8ed76a530c8c3cc1d49eecf9d250c/addons/survey/views/survey_templates.xml#L27

we check if we have to display the progression or not, in case of consulting results it doesn't make sense displaying it.
That means the check made in `t-if` was not enough in this case and a condition is missing.

Specification
=============

to solve the issue the condition `not survey_form_readonly` was added

opw-2821478

### CODE REVIEW

File: addons/survey/views/survey_templates.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
             <div class="py-3 m-0 p-0 text-right">
                 <div class="o_survey_progress_wrapper d-inline-block pr-5 text-left">
                     <t t-call="survey.survey_progression"
-                        t-if="survey and survey.questions_layout != 'one_page' and answer and answer.state == 'in_progress' and (not question or not question.is_page)">
+                        t-if="survey and survey.questions_layout != 'one_page' and answer and answer.state == 'in_progress' and (not question or not question.is_page) and (not survey_form_readonly)">
```
Comment (abz-odoo) [inline addons/survey/views/survey_templates.xml:27]:
```suggestion
                        t-if="survey and survey.questions_layout != 'one_page' and answer and answer.state == 'in_progress' and (not question or not question.is_page) and not survey_form_readonly">
```
                        
I think the parentheses are useless

**Review by abz-odoo — CHANGES_REQUESTED** [review]

File: addons/survey/views/survey_templates.xml (hunk 1 above)
Comment (abla001) [reply addons/survey/views/survey_templates.xml]: that's true, I removed them, thank you :pray:

**Review by abz-odoo — APPROVED** [review]
LGTM
Lets see @awa-odoo's feedback

**Review by awa-odoo — APPROVED** [review]
Hello @abla001 @abz-odoo 

Glad to see the suggested fix is working!
LGTM 🙂

**Review by nle-odoo — APPROVED** [review]
looks good to me

alternative would be a condition based on if we have a page (but it would need to depend on `survey.questions_layout == 'page_per_section'` complicating this)

the link in commit message can become wrong after some days, you should go to the link and press "y" (see [source code keyboard shortcut](https://docs.github.com/en/get-started/using-github/keyboard-shortcuts#source-code-browsing)) to get canonical link in address bar

**Comment by abla001** [comment]
> looks good to me
> 
> alternative would be a condition based on if we have a page (but it would need to depend on `survey.questions_layout == 'page_per_section'` complicating this)
> 
> the link in commit message can become wrong after some days, you should go to the link and press "y" (see [source code keyboard shortcut](https://docs.github.com/en/get-started/using-github/keyboard-shortcuts#source-code-browsing)) to get canonical link in address bar

ah I didn't know that, I updated the commit message, thank you :pray:

**Comment by nle-odoo** [comment]
robodoo r+

**Pull request closed without merging**